lib/post_revisor.rb

```diff
@@ -601,4 +604,43 @@ def successfully_saved_post_and_topic
     @post_successfully_saved && !@topic_changes.errored?
   end
 
+  def revise_likes(deleted_post_ids, acting_user)
+    return if deleted_post_ids.blank?
+
+    already_liked_users = PostAction.where(post_id: @post.id, post_action_type_id: PostActionType.types[:like]).pluck(:user_id)
+
+    args = {}
+    args[:message] = "Merging posts"
+    args[:take_action] = true
+    args[:flag_topic] = true
+
+    user_with_transferred_likes = Set.new
+    PostAction.where(post_id: deleted_post_ids, post_action_type_id: PostActionType.types[:like]).find_in_batches do |group|
+      group.each do |post_action|
```

Comment:
I would personally prefer 4 SQL statements here instead of the looping, there are some odd cases here in the current pattern where stuff will crash out like missing user action and so on. SQL avoids needing the extra gates and is generally faster.

Reply:
Can you explain how you would remove the looping using SQL statements?

From what I understand what you mean is removing below and writing corresponding sql for it:
```
user_action_like = UserAction.where(target_post_id: deleted_post_id, user_id: post_action.user_id, action_type: UserAction::LIKE, target_topic_id: deleted_post.topic_id).first
user_action_like.update_column(:target_post_id, @post.id)
```
But how does it remove looping?